### ISSUE TITLE
Display of S-101 ENCs with different FCs and PCs

### ISSUE THREAD

**Issue by skjeves** [iho-ohi/S-164-Sub-Group#1]
Seamless display of data side by side or when zooming in/out (overlapping or different usage datasets). 

For S-100 ECDIS, the situation could arise that on screen one part of the screen contains an S-101 dataset using one version of FC/PC, and on the other part of the screen an S-101 dataset is using another version of the FC/PC. There should be test(s) ensuring those situations are properly captured and displayed.

There is also the situation where you look at a dataset using one version of FC/PC, and when zooming in/out another dataset using another version of FC/PC could be displayed - also take into consideration that this situation could occur only partly on the screen .

Also related to how many versions of catalogues for one S-100 product should an ECDIS support?

**Comment by kusala9** [comment]
Agree, this could be done within some of the existing "PowerUp" tests which can use different versions of the FC/PC and we could add to the instructions the correct pan/zoom instructions. 

ECDIS must support at least two versions but there's no guidance for how many as a maximum. This maybe should be documented in S-98 Annex C (even as a recommendation?). One for discussion.

**Comment by rmalyankar** [comment]
S-98 Annex C C-21.1 (Multiple product versions and portrayal) discusses the implications for portrayal catalogues of requiring ECDIS to support multiple PS versions. If the "consequences" in that clause are taken into account when updating product specifications (including portrayal catalogues), then the ECDIS need maintain only the latest PCs for new Editions of product specifications (the concept in Annex C C-21.1 is that PCs for Revisions and Clarifications should be able to handle datasets for earlier Revisions and Clarifications within the same Edition number of the Product Specification).

So the answers to
> ... how many versions of catalogues for one S-100 product should an ECDIS support?
could be
1. It need support only the latest PC for the active Editions. This would also mitigate though not entirely solve the question of visual differences for different versions.
2. For FCs - maybe "as many Edition+Revision combinations as exist for active Editions"? (Clarifications are supposed to be backward compatible.)

So I think this issue has three sides:
1. What should S-98 Annex C say about it (cf. Annex C Clause C-21.1). This discussion should probably be in the [interoperability repository](https://github.com/iho-ohi/98-interoperability/issues).
2. What related tests should S-164 define?

This wouldn't entirely solve the issues raised in the top comment by @skjeves but it would make the problems smaller and easier to manage.

**Comment by DavidGrant-NIWC** [comment]
Currently, S-101 demonstrates that revisions may modify the data model, as evidenced by the changes in 1.0.2 and 1.1.0. S-101 1.2.0 will further modify the data model.
* The latest PC for a given revision should be used.

However, that only applies to the rules and alerts. For other components of the PC the latest version of the component should always be used, regardless of the edition/revision of the dataset or catalogs: pixmaps, colors, symbols, linestyles, areafills, fonts, and context parameters. In the case where a component is not present in the latest catalog (it's retired/removed) the ECDIS will need to use the last available version.

Assuming we have the following catalogs available:
FC 1.0.2, FC 1.1.0
PC 1.0.2, PC 1.0.3
PC 1.1.0, PC 1.1.1, PC 1.1.2

The mapping from dataset to catalog is (the last dataset is unusable):
|Dataset|FC|PC|
|:---|:---:|:---:|
|1.0.0|1.0.2|1.0.3|
|1.1.0|1.1.0|1.1.2|
|1.2.0|-|-|

Viewing groups, layers, display modes, and display planes are more complicated. See [PSWG issue 16](https://github.com/S-101-Portrayal-subWG/Working-Documents/issues/16). It should not be permitted to move an existing viewing group between viewing group layers, and there should be a check to ensure that different products will not put the same viewing group in different viewing group layers.

A similar restriction must be applied to display modes (the assignment of a viewing group layer to a display mode must always be consistent across all PC's and all product types).

**Comment by rmalyankar** [comment]
I think we're going to have to treat pre-2.0 versions of product specifications as a special case. If we don't we will never be able to simulate or test the management of new or multiple Editions on ECDIS until after real 2.0 Editions are released. How to implement such special treatment without contaminating the handling of post-2.0 editions is a good question.

**Comment by DavidGrant-NIWC** [comment]
We need to be able validate that we can execute an update cadence using the agreed upon model. If all of our test artifacts update the revision instead of the edition, then we haven't fully vetted the process. Our test bed currently uses the revision to associate catalogs because that is the only mechanism that works with existing test artifacts.
